### DISCUSSION TITLE
Editing Custom fields

### DISCUSSION THREAD

**Discussion by mradrianmarsh** [go-jira/jira#501] [Q&A]
My Jira Cloud instance has what I think are custom fields (Due Date and Start Date) on EPICs.
I have list of updates I want to loop through the cli.

But I can't see how to find the custom field names to use.   Would the cli tool be usable here ?

**Comment by sirhc** [comment]
This is something I have to deal with a lot and I have a few places where I've been able to read custom fields and update them.

First, finding them can be tricky, depending on how unique the name is.

```
# jira fields | fx
```

(The [`fx`](https://fx.wtf/) command is handy for exploring JSON, but `less` would work just as well here.)

I'll then cross-reference that with an issue where I know what the value is.

```
# jira view ISSUE-1 -t fields | fx
```

Once I know the field, I create a template that uses it. For example, we have several required custom fields when transitioning an issue to a state indicating that it's ready for a code review. So I have something like this in `~/.jira.d/templates/code-review`:

```yaml
fields:
  customfield_12345: # code reviewer
    emailAddress: {{ .overrides.reviewer }}
  # etc.
```

Then I can transition the issue thusly,

```
# jira transition 'Code Review' ISSUE-1 -t code-review -o reviewer=reviewer@example.com
```

Similarly for viewing an issue.

```
# jira view ISSUE-1 -t fields | jq '.fields.customfield_12345.emailAddress'
```

Or provide a custom template that includes the custom field (or modify the existing view template).
